### ISSUE TITLE
Chase is listed as having 2 factor auth enabled, yet it doesn't have this feature available

### ISSUE THREAD

**Issue by ChiefPoints** [2factorauth/twofactorauth#1548]
Under "Banking", the list states that "Chase 2FA is always enabled" and it couldn't be further from the truth. It's only enabled when creating the account, which leaves you completely unprotected if your computer/laptop is stolen. I just went through this same discussion with a "customer service" rep at Chase. For 2 factor auth to be useful, it must be enabled for ALL logins - otherwise a stolen laptop gives away the keys to the kingdom.
To be clear, Chase only attempts to verify a user when the account is first created, subsequent logons are not protected with 2 factor auth. They can call it whatever they want, but this isn't 2 factor authentication.

**Comment by mxxcon** [comment]
That is not completely correct. See  #263 and #644

**Comment by ChiefPoints** [comment]
I've seen those comments and still stand by my statement. If someone knows your password and has access to your computer, they can get into your account - exactly the kind of thing that 2 factor auth is supposed to prevent. If Chase wants to go a step further and offer the users the choice to not require 2-factor auth for a device after you've logged in, that's fine and that's on the user, but it should never default to a setting like that - simply unacceptable.

**Comment by jamcat22** [comment]
Hello @ChiefPoints!

That's how most of the Cloud Syncing and Financial/Banking services we have listed work.

Take Dropbox for example. If someone had access to your computer, they could technically access all of your data without having to use 2FA, even if you unlink your account remotely.

While I do agree that it should be an option to require 2FA on each and every use, I think it is better that they require and even have the feature in the first place. So many financial institutions don't even have the feature, and very few enforce it for everyone. My guess would be that the tradeoff Chase took was to make it required and as simple as possible for all users, instead of making it optional and complicated. Check out the [many](https://support.mozilla.org/questions/1010132) [different](https://www.directly.com/p/chase-why-do-i-have-to-get-an-identification-code-everytime-i-try-to-log-into-my-accounts-35860) [questions](https://discussions.agilebits.com/discussion/27877/accessing-chase-bank-on-android-phone-always-requires-new-online-id-code) [from](http://forums.windowscentral.com/windows-phone-8-1/349109-dilemma-chase-mobile.html) [people](http://forums.mozillazine.org/viewtopic.php?f=38&t=2828753) when Chase asks for a code on a device they already logged in to. Making the process as painless as possible, while still increasing security is the only way Chase can convince all of its users that requiring 2FA is a good thing.

Their system is also better than many banks that require 2FA, as many that require it only need it for certain transactions, not to login.

Furthermore, a simple password change after your computer has been stolen should automatically invalidate any old auth tokens.

I'd also say that protecting logins from a known computer isn't really the main purpose of 2FA. It's main purpose is to prevent remote access, password cracking, and social engineering. If a laptop is stolen (even with 2FA enabled), it's always best to reset your passwords.

In conclusion, while I agree that this should be an option, the absence of this option doesn't warrant removing Chase, as Chase still fits our definition of 2FA.

**Comment by ChiefPoints** [comment]
Changing your password isn't an option when you are at work, someone breaks into your house and accesses your devices while there - an act that is beginning to happen with regularity.  If this isn't a scenario people consider when discussing 2FA, they're missing the boat. It's my belief that if entities like Chase are allowed to claim what they offer is true 2FA, then they'll never come around to where they need to be. While I do agree that what Chase provides is better than nothing, "nothing" isn't even an option in today's threat environment, kind of like saying it's "better than a sharp stick in the eye". The purpose of 2FA is to prevent someone from logging into your accounts without meeting the proper requirements, wherever that logon attempt may originate from. The way they have it set up leaves too many open doors to be truly useful. I understand that you say Chase fits your definition of 2FA, I just believe that definition needs some tightening up to force these entities into proper action.

**Comment by jamcat22** [comment]
In that case, 2FA wouldn't be at fault. One of the possible factors for 2FA is possession. If you lose your laptop, and it's not protected with encryption, then one of your factors (possession) is compromised. In Chase's case, your second factor is your phone/email, then after you authenticate the first time, your second factor becomes your computer.

**Comment by Carlgo11** [comment]
Just out of curiosity, do they verify using cookies?
If so, one could just remove all cookies after closing the browser. I do that.

I'm not suggesting everyone should do that. I'm simply giving anyone interested a solution if they feel they're at more risk than the average Joe.

Also, [Encryption](https://veracrypt.codeplex.com/).

**Comment by ChiefPoints** [comment]
jamcat22 - No, 2FA wouldn't be at fault - Chase would be for not allowing it to be set up correctly. The second factor you mention only comes into play when adding a device to your account for the first time, and does absolutely nothing to protect you in this scenario. When you say that "after you authenticate for the first time, your second factor becomes your computer", do you realize that you just assigned the computer as both factors? Do you really think that someone couldn't figure out your password if they had your laptop in their possession? How many users do you think completely encrypt all their drives, betting It's a pretty small number. It's a small thing to allow 2FA on all logons, I really don't understand the push back against it when it offers so much more protection.

Carlgo11 - Yes, they do verify with cookies, and deleting them is an option - but again, not many ppl are willing to do that, and even fewer will go through the trouble of setting up which cookies to remove and which to leave. The vast majority of users out there are oblivious to the handling of cookies. Personally, I do clear out cookies, encrypt and take other precautions, but as a security guy I can't assume that all users can or will do that. Those of us that have the knowledge can easily protect ourselves - the ones without the knowledge are the ones I worry most about. 

In any case, I've said my piece. If everyone used proper passwords, thieves wouldn't be able to access their accounts so easily. Security features like 2FA are there to help protect those that think 123456 is a good password and use the same password for all accounts. As long as we allow entities like Chase get away with a half-hearted implementation of the technology, things won't get better. On the same note, users must take the same initiative to learn out to protect themselves, then maybe we can all meet in the middle somewhere.

**Comment by christianjuth** [comment]
I'd like to point out that while Chase technically has 2FA enabled there is a big design flaw. As mentioned in this thread Chase will automatically set a computer as a trusted. I have tried clearing cookies and local storage for Chase, but it still trusts the device. Someone at Chase told me they use the ip address to see if you have verified your identity before. This means if you work at a coffeeshop it will verify your identity for anyone who tries to log in at that same coffeeshop. I even tried using a VPN and changing my location and it still seems to stay verified weather I am tunneled to California or NYC.

**Comment by mxxcon** [comment]
>  Someone at Chase told me they use the ip address to see if you have verified your identity before.

Who? Is there documented proof of this?

>This means if you work at a coffeeshop it will verify your identity for anyone who tries to log in at that same coffeeshop.

This doesn't make sense.

I simply opened Incognito window and Chase initiated their 2fa process.
I opened Safari browser on the same computer and it initiated 2fa process.
So...there's that...

**Comment by developer1011** [comment]
<blockquote><p>I simply opened Incognito window and Chase initiated their 2fa process.</p>
</blockquote>
  
No it isn't I opened Incognito window and logged in to chase. It did not ask for 2fa

**Comment by mxxcon** [comment]
Well, I just tried logging in from Incognito was prompted with this.

![clipboard01](https://user-images.githubusercontent.com/847140/33561941-032992ec-d8e2-11e7-8ecb-45c0f72629e5.png)
![clipboard02](https://user-images.githubusercontent.com/847140/33561942-034c296a-d8e2-11e7-95d6-68723eae3642.png)

**Comment by schwit61** [comment]
I have the same problem. Multiple browsers(IE, FireFox and Chrome) all do 2FA first time login, but never do 2FA after this. Chase tech support says a cookie is set and all I need to do is clear cookies to get the 2FA back. That didn't work or they are setting a permanent cookie.

**Comment by jadacyrus** [comment]
I’d actually like to weigh in here as someone who works in computer security. Chase definitively does not have MultiFactor authentication. 

MultiFactor authentication is:
- Something you are (biometric)
- Something you have (token)
- Something you know (password)

Combine two of these factors for authentication for each login and you have real MultiFactor authentication. What chase offers does not make the cut. 

Email and SMS for “new” devices is not true multifactor. SMS has been proven to be an insecure method of transmission for codes and almost all security guidelines warn against using “SMS two factor” because it can be intercepted easily. Besides that many of the assumptions we make about SMS being a second factor are incorrect:

- People may share phone numbers
- Phone numbers may change
- People who work at phone company can see messages

Email has a lot of the same problems and is not a true second factor. If an attacker has my password they may also have the same password for my EMail account.  In that case it’s proven that only one factor (password) can get access to a Chase account.

If Chase was serious about security they would allow users to enable TOTP/HOTP authentication for each login for true multifactor authentication.

The argument that TwoFactor auth exists merely to protect people with bad passwords is the worst argument i’ve ever heard. Companies get breached regularly and that user data often ends up in the hands of malicious people. There are also often scenarios where you may not completely trust the network you are connected to (public network / or traffic being MITMd via corporate proxy).

**Comment by Storm1** [comment]
I couldn't agree more. I have a lot of money in Chase and the account sign-in is not secure.  I wish they would add Yubikey support. This thread started years ago and I see similar discussions on the web yet not a thing has been improved.

**Comment by techsone** [comment]
Everyone should tweet Chase and continue to complain about this.  With all type of security breaches hitting the news each day, they need to listen to what consumer want to better protect themselves.

**Comment by Palisand** [comment]
Through their website, Chase offers `Extra security at sign-in` under `Profile & Settings` -> `AccountSafe™️`.
> Every time you sign in on a browser, we'll send you a one-time code to use with your password to verify your identity.

**Comment by zecanard** [comment]
I can confirm that Chase asks me for an SMS code every time I log in.

**Comment by clixx52** [comment]
A one-time code on every sign-in is still not properly secure if it is transmitted by email or SMS.  They need to provide support for TOTP/HOTP.

**Comment by TheCodingWaffle** [comment]
> Through their website, Chase offers `Extra security at sign-in` under `Profile & Settings` -> `AccountSafe™️`.
> 
> > Every time you sign in on a browser, we'll send you a one-time code to use with your password to verify your identity.

Looked for this feature for a solid 30 minutes and can say It doesn't exist for me. I have 3 accounts with Chase and this is concerning. Further looking into this it's a feature only offered to some customers at this moment. Why is my steam account more secure then my banking account?!

**Comment by alimas7** [comment]
> A one-time code on every sign-in is still not properly secure if it is transmitted by email or SMS. They need to provide support for TOTP/HOTP.

Exactly.

Chase should be embarrassed for not knowing technology (or having competent technical employees).

TOTP/HOTP implementation is very simple.

**Comment by bartenbach** [comment]
I really don't have a dog in this fight other than to agree that Chase has the worst possible implementation of 2FA that I have ever seen. I can't even find where it's configurable. It might be buried in the settings somewhere but it's not under "sign in and security" which would have made sense.

I also like how you have to re-enter your password again on the page where you enter the code from the text message. I'm not sure what the use case for that is. Someone got lucky guessing the password one time and *also has your phone* but can't remember what they just typed on the previous page? Most browsers just autofill it which completely defeats the purpose, but even if they didn't, you *literally just typed it on the previous page*.

**Comment by zecanard** [comment]
I believe two-factor authentication is enabled automatically for every phone number and email address you add to **Profile & Settings** → **Personal Details** → **Phone** and **Email**.

It appears that every phone number and email address must be associated with at least one Chase account. So in my case, both of my phone numbers, as well as my email address, are automatically associated with my one and only Chase account, and can be selected for 2FA during login. However, in settings, my secondary phone number does get the option: **Add accounts or make primary >**. Clicking that link gives me the option to make it the primary contact method for that account, but the option to deselect my Chase account is grayed out.

As for having to enter your password twice during login, it’s a fairly recent change (a few months ago?), and it has left me very confused. All it does is slow down the login process, while offering no additional protection.

**Comment by GoldenChrysus** [comment]
> > Through their website, Chase offers `Extra security at sign-in` under `Profile & Settings` -> `AccountSafe™️`.
> > > Every time you sign in on a browser, we'll send you a one-time code to use with your password to verify your identity.
> 
> Looked for this feature for a solid 30 minutes and can say It doesn't exist for me. I have 3 accounts with Chase and this is concerning. Further looking into this it's a feature only offered to some customers at this moment. Why is my steam account more secure then my banking account?!

Found this, but it's buried and nowhere near where it should be.

(Sign in to Chase) → Security & privacy (in top navigation bar) → Add extra security when you sign in using a browser (way at the bottom under "Ways you can be more secure" under "Resources").

This takes you to Profile & Settings on a page that can't be accessed anywhere from within Profile & Settings itself...

**Comment by tuxedonight** [comment]
> > > Through their website, Chase offers `Extra security at sign-in` under `Profile & Settings` -> `AccountSafe™️`.
> > > > Every time you sign in on a browser, we'll send you a one-time code to use with your password to verify your identity.
> > 
> > 
> > Looked for this feature for a solid 30 minutes and can say It doesn't exist for me. I have 3 accounts with Chase and this is concerning. Further looking into this it's a feature only offered to some customers at this moment. Why is my steam account more secure then my banking account?!
> 
> Found this, but it's buried and nowhere near where it should be.
> 
> (Sign in to Chase) → Security & privacy (in top navigation bar) → Add extra security when you sign in using a browser (way at the bottom under "Ways you can be more secure" under "Resources").
> 
> This takes you to Profile & Settings on a page that can't be accessed anywhere from within Profile & Settings itself...

Dude thank you so much for this; I created an account on here just to thank you lol!

**Comment by TheCodingWaffle** [comment]
> > > Through their website, Chase offers `Extra security at sign-in` under `Profile & Settings` -> `AccountSafe™️`.
> > > > Every time you sign in on a browser, we'll send you a one-time code to use with your password to verify your identity.
> > 
> > 
> > Looked for this feature for a solid 30 minutes and can say It doesn't exist for me. I have 3 accounts with Chase and this is concerning. Further looking into this it's a feature only offered to some customers at this moment. Why is my steam account more secure then my banking account?!
> 
> Found this, but it's buried and nowhere near where it should be.
> 
> (Sign in to Chase) → Security & privacy (in top navigation bar) → Add extra security when you sign in using a browser (way at the bottom under "Ways you can be more secure" under "Resources").
> 
> This takes you to Profile & Settings on a page that can't be accessed anywhere from within Profile & Settings itself...

For anyone else not able to find this in the web, I found it in the Android app every easily since the web portal has changed significantly since this solution was posted.
